Tcp router PR

atulkc opened the pull request:
This PR is for adding tcp router functionality to lattice.
- It adds haproxy, router-configurer, tcp-emitter components to lattice
- Upstart and compile scripts changed to take care of these new components
- Vagrantfile modified to install incrontab on lattice-brain
- AWS terraform modified to install incrontab on lattice-brain
- Updated `ltc create` to take `--tcp-routes` flag to let developers configure tcp routes for their apps
- Added integration test for tcp routing support testing

Regards,
@atulkc and @fordaz 

PS:  This PR replaces  https://github.com/cloudfoundry-incubator/lattice/pull/171, as indicated by David Wadden

davidwadden commented:
Thanks for getting this right =)  Sorry for the delay, but we're planning to cut a **v0.3.1** patch release soon (within a day or two) to address some things from **v0.3.0** going out last week before we merge tcp-router functionality.  I've tested this a bit, and it looks good though.  I'll probably reach out in the next day or two to go through the final rebase once we're ready to merge.

atulkc commented:
Sure...let us know when you are ready to merge in the changes.

fordaz committed:
Package tcp-emitter with the lattice tarball.

  - Created upstart script for tcp-emitter
  - Added a dependency on the cf-routing-release

[#98237046]

Signed-off-by: Chris Piraino <cpiraino@pivotal.io>

crhino committed:
Install router-configurer and haproxy onto lattice and run them.

[#98237152]

Signed-off-by: Fermin Ordaz <fermin.ordaz@gmail.com>

fordaz committed:
Enable reloading of haproxy config to route tcp traffic.

[#98237152]

Signed-off-by: Chris Piraino <cpiraino@pivotal.io>

fordaz committed:
Update development scripts to include TCP routing.

[#98237152]

fordaz committed:
Fixed haproxy upstart/incrontab reload issues.

[#98237152]

leochu committed:
Add tcp-route flag to ltc

[#94628548]

Signed-off-by: Fermin Ordaz <fermin.ordaz@gmail.com>

fordaz committed:
Support of the --tcp-routes flag in the create CLI command

[#94628548]

atulkc committed:
Verify port in tcp-routes to be within valid port range

  + modify help message to show comma separated port pair in tcp-routes

[#94628548]

Signed-off-by: Fermin Ordaz <fermin.ordaz@gmail.com>

fordaz committed:
Update ltc.md with description of the new tcp-routes flag.

[#94628548]

Signed-off-by: Atul Kshirsagar <atul.kshirsagar@gmail.com>

atulkc committed:
Add tcp-routing integration test.

[#98237152 #98322946]

Signed-off-by: Fermin Ordaz <fermin.ordaz@gmail.com>

atulkc committed:
Fixed upstart script to respawn and correctly handle reload.

[#100228278]

Signed-off-by: Matthew Boedicker <mboedicker@pivotal.io>

fordaz committed:
Use upstart-file-bridge in place of incrontab

[#100228278]

Signed-off-by: Atul Kshirsagar <atul.kshirsagar@gmail.com>

davidwadden committed:
Correct newlines in TCP Routes output

[#101089176]
